src/gluonts/mx/model/estimator.py

```diff
@@ -170,27 +172,34 @@ def train_model(
         shuffle_buffer_length: Optional[int] = None,
         cache_data: bool = False,
     ) -> TrainOutput:
+
         transformation = self.create_transformation()
 
-        transformed_training_data = transformation.apply(training_data)
+        with env._let(max_idle_transforms=max(len(training_data), 100)):
+            transformed_training_data = transformation.apply(training_data)
+            if cache_data:
+                transformed_training_data = Cached(transformed_training_data)
```

Comment:
Here we are setting the `max_idle_transforms` when applying the transformation to the data.